### PR TITLE
[Octo-Bot] Adding new Packet Generator Bot

### PR DESCRIPTION
**Pull request by krusagiz**
Adds a new Octo-Bot that provides packet generation capabilities written in Python

### CODE REVIEW

**Comment by ariscahyadi** [comment]
I run this command `sudo docker run packet-generator-bot:latest -r captures/http.cap` as mentioned in the documentation.

But I got this error:

```
Iteration #1
Traceback (most recent call last):
  File "main.py", line 41, in <module>
    main()
  File "main.py", line 36, in main
    generateReplay(args)
  File "/ReplayGenerator.py", line 65, in generateReplay
    generator.sendPackets(outfile=args.outfile)
  File "/PacketGenerator.py", line 95, in sendPackets
    print ('sending %i packets a sec...' % send_rate)
TypeError: %i format: a number is required, not NoneType

```

**Comment by krusagiz** [comment]
> I run this command `sudo docker run packet-generator-bot:latest -r captures/http.cap` as mentioned in the documentation.
> 
> But I got this error:
> 
> ```
> Iteration #1
> Traceback (most recent call last):
>   File "main.py", line 41, in <module>
>     main()
>   File "main.py", line 36, in main
>     generateReplay(args)
>   File "/ReplayGenerator.py", line 65, in generateReplay
>     generator.sendPackets(outfile=args.outfile)
>   File "/PacketGenerator.py", line 95, in sendPackets
>     print ('sending %i packets a sec...' % send_rate)
> TypeError: %i format: a number is required, not NoneType
> ```

I have fixed the bug. The print was actually dangling code that I forgot to remove.

File: Octo-Bot/packet-generator-bot/bot/PacketGenerator.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import time
+
+from scapy.all import *
+
+
+class PacketGenerator():
+
```
Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/bot/PacketGenerator.py:7]: Can you a little documentation of each function?

Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/bot/PacketGenerator.py:7]:
```suggestion
  """ 
  Class to generate packet
  input : XXXX
  """
```

File: Octo-Bot/packet-generator-bot/bot/ReplayGenerator.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+from scapy.all import *
+
+
+from PacketGenerator import *
+from helper.ReplayHelper import *
+from utils.TimeUtils import *
+
```
Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/bot/ReplayGenerator.py:7]: Can you a little documentation for this function?
Including expected inputs/arguments.

File: Octo-Bot/packet-generator-bot/bot/helper/ReplayHelper.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from scapy.utils import rdpcap
+from os import path
+
+
```
Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/bot/helper/ReplayHelper.py:4]: Can you a little documentation for this function?
Including expected inputs/arguments.

File: Octo-Bot/packet-generator-bot/bot/main.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import argparse
+
+from os import system
+
+# Internal Imports
+from utils.RandNetParamsGen import *
+from ReplayGenerator import *
+
+
```
Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/bot/main.py:9]: Can you a little documentation for this function?

File: Octo-Bot/packet-generator-bot/bot/utils/RandNetParamsGen.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import random
+
```
Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/bot/utils/RandNetParamsGen.py:2]: Can you a little documentation for this function?
Including expected inputs/arguments.

File: Octo-Bot/packet-generator-bot/bot/utils/TimeUtils.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+
+MIN_IN_SECS = 60
+HOUR_IN_SECS = 3600
+DAY_IN_SECS = 86400
+
```
Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/bot/utils/TimeUtils.py:5]: Can you a little documentation for this function?
Including expected inputs/arguments.

File: Octo-Bot/packet-generator-bot/test.sh [hunk 7]
```diff
@@ -0,0 +1,15 @@
+chmod a+x *.sh &&
```
Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/test.sh:1]: Can you a little documentation for this function?
What are the expected results?

File: Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import unittest
+
+from scapy.all import *
+
+from helper.ReplayHelper import *
+
+
```
Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py:7]: Can you a little documentation for this function?
Including expected inputs/arguments.

File: Octo-Bot/packet-generator-bot/tests/test_utils_timeutils.py [hunk 9]
```diff
@@ -0,0 +1,49 @@
+import unittest
+
+from utils.TimeUtils import *
+
```
Comment (ariscahyadi) [inline Octo-Bot/packet-generator-bot/tests/test_utils_timeutils.py:4]: Can you a little documentation for this function?
Including expected inputs/arguments.

File: Octo-Bot/packet-generator-bot/bot/PacketGenerator.py (hunk 1 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/bot/PacketGenerator.py:23]: I have added documentation to most of the main functions!

Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/bot/PacketGenerator.py:23]: Added!

File: Octo-Bot/packet-generator-bot/bot/ReplayGenerator.py (hunk 2 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/bot/ReplayGenerator.py:7]: Added!

File: Octo-Bot/packet-generator-bot/bot/helper/ReplayHelper.py (hunk 3 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/bot/helper/ReplayHelper.py:4]: Added!

File: Octo-Bot/packet-generator-bot/bot/main.py (hunk 4 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/bot/main.py:10]: Added!

File: Octo-Bot/packet-generator-bot/bot/utils/RandNetParamsGen.py (hunk 5 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/bot/utils/RandNetParamsGen.py:2]: Added!

File: Octo-Bot/packet-generator-bot/bot/utils/TimeUtils.py (hunk 6 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/bot/utils/TimeUtils.py:5]: Added!

File: Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py (hunk 8 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py:7]: Added!

File: Octo-Bot/packet-generator-bot/tests/test_utils_timeutils.py (hunk 9 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/tests/test_utils_timeutils.py:4]: Added!

File: Octo-Bot/packet-generator-bot/bot/PacketGenerator.py (hunk 1 above)
Comment (ariscahyadi) [reply Octo-Bot/packet-generator-bot/bot/PacketGenerator.py:23]: Thank @krusagiz , I will make as `resolved`

File: Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py (hunk 8 above)
Comment (ariscahyadi) [reply Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py:7]: I didn't any documentation for this class?

File: Octo-Bot/packet-generator-bot/tests/test_utils_timeutils.py (hunk 9 above)
Comment (ariscahyadi) [reply Octo-Bot/packet-generator-bot/tests/test_utils_timeutils.py:4]: I didn't any documentation for this class?

**Comment by ariscahyadi** [comment]
I also have a problem when running the test script.

```
arisdia1@Octo-Bot/packet-generator-bot$ ./test.sh 
Traceback (most recent call last):
  File "bot/test_helper_replayhelper.py", line 3, in <module>
    from scapy.all import *
ImportError: No module named scapy.all
Traceback (most recent call last):
  File "bot/test_utils_timeutils.py", line 3, in <module>
    from utils.TimeUtils import *
ImportError: No module named utils.TimeUtils

Please check if the tests succeeded!
```

File: Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py (hunk 8 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py:7]: Some documentation was added to the main test.sh shell script. While slightly verbose, I will add it to this script as well.

File: Octo-Bot/packet-generator-bot/tests/test_utils_timeutils.py (hunk 9 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/tests/test_utils_timeutils.py:4]: Some documentation was added to the main test.sh shell script. While slightly verbose, I will add it to this script as well.

Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/tests/test_utils_timeutils.py:4]: I have added a simple note as documentation in the test python scripts.

File: Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py (hunk 8 above)
Comment (krusagiz) [reply Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py:7]: I have added a simple note as documentation in the test python scripts.

**Comment by krusagiz** [comment]
> I also have a problem when running the test script.
> 
> ```
> arisdia1@Octo-Bot/packet-generator-bot$ ./test.sh 
> Traceback (most recent call last):
>   File "bot/test_helper_replayhelper.py", line 3, in <module>
>     from scapy.all import *
> ImportError: No module named scapy.all
> Traceback (most recent call last):
>   File "bot/test_utils_timeutils.py", line 3, in <module>
>     from utils.TimeUtils import *
> ImportError: No module named utils.TimeUtils
> 
> Please check if the tests succeeded!
> ```
This happens when the bot is ran using python2, as it is made for python3 only. Please alias `python` to `python3`. I will update the test.sh script to force the usage of python3 as well.

File: Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py (hunk 8 above)
Comment (ariscahyadi) [reply Octo-Bot/packet-generator-bot/tests/test_helper_replayhelper.py:7]: Ok!

**Pull request merged**